sdk/keyvault/Azure.Security.KeyVault.Shared/src/ChallengeBasedAuthenticationPolicy.cs

```diff
@@ -52,30 +43,24 @@ protected async Task PreProcessAsync(HttpMessage message, ReadOnlyMemory<HttpPip
                 // As a result, before we know the auth scheme we need to avoid sending an unprotected body to Key Vault.
                 // We don't currently support this enhanced auth scheme in the SDK but we still don't want to send any unprotected data to vaults which require it.
 
-                RequestContent originalContent = message.Request.Content;
+                message.SetProperty(KeyVaultStashedContentKey, message.Request.Content);
```

Comment:
I would still prefer if we explicitly send the request right here instead of going through the base class and back around to the `AuthenticateRequestOnChallengeAsync`

Reply:
I can see both arguments - either we provide sub-classes the ability to do anything they want with the pipeline in `AuthenticateRequest`, which could be abused, or we rely on some level of cooperation between the sub-class and the base which is based on some guarantees (`AuthenticateRequest` gets called before base.Processs and `AuthenticateRequestOnChallengeAsync` always gets called in response to a challenge.

@schaabs Do you have any strong opinions here?

Reply:
I prefer it the way it's written now. If we send the message here we end up duplicating the code to handle the challenge and the essentially short circuit the base class logic. I think the base class abstraction is clean, you essentially have two opportunities to authenticate the request, before it is sent, and if an authentication challenge is received.  Also I think this subclass in this case is conforming to that conceptual model as it can't authenticate the request originally it doesn't add an Authorization header and allows the request to be sent relying on the service to respond with an WWW-Authenticate challenge to give the details needed to authenticate. I agree that removing the body and stowing it on the message and readding it after the challenge might seem a bit like spooky action at a distance, but I don't think it's too bad since it's stashed and used in the same class.